### PR TITLE
Ensure NJOY output goes to specified output directory

### PR DESCRIPTION
**Pull request by paulromano**
This PR solves an issue that I've been observing for some time. One of our scripts for data generation ([generate_endf71.py](https://github.com/openmc-dev/data/blob/master/generate_endf71.py)) runs many instances of NJOY in parallel when generating HDF5 files. However, when I've run this script on a large node with many cores, I've observed that sometimes one of the tasks will fail randomly. After looking at it a little closer, I realized that it's because of how ACE files are being written out. When multiple temperatures are specified, the ACE files were not being written to the specified output directory, and so if there were many parallel jobs running in the same directory, they would sometimes clobber each other, resulting in a missing file.

### CODE REVIEW

**Comment by Shimwell** [comment]
Interesting, I think I have noticed this. Is this the reason for the try: except: part in here? [https://github.com/openmc-dev/data/blob/6d9fc7cb7c2ac429c978a563a7c4407f564d5a3b/generate_endf71.py#L65](url)

**Comment by paulromano** [comment]
Well, the errors get caught by that try/except block in the generate_endf71.py script, but it's really intended for evaluations that _actually_ fail (for reasons other than bugs in OpenMC itself).

File: openmc/data/njoy.py [hunk 1]
```diff
@@ -573,21 +580,18 @@ def make_ace_thermal(filename, filename_thermal, temperatures=None,
         commands += _THERMAL_TEMPLATE_ACER.format(**locals())
 
         # Indicate tapes to save for each ACER run
-        tapeout[nace] = fname.format(ace, temperature)
-        tapeout[ndir] = fname.format(xsdir, temperature)
+        tapeout[nace] = output_dir / fname.format(ace, temperature)
+        tapeout[ndir] = output_dir / fname.format(xsdir, temperature)
```
Comment (amandalund) [inline openmc/data/njoy.py]: Should this be `"xsdir"` now that the default value of `xsdir` is `None`?

Comment (paulromano) [reply openmc/data/njoy.py]: Yes, you're absolutely right. Will get this fixed; thanks!

**Pull request merged**